src/tm_devices/drivers/pi/signal_generators/awgs/awg.py

```diff
@@ -0,0 +1,494 @@
+    # Magic Methods
+    ################################################################################################
+    def __init__(self, awg: AWG, channel_name: str) -> None:
+        """Create an AWG source channel.
+
+        Args:
+            awg: An AWG.
+            channel_name: The channel name for the AWG source channel.
+        """
+        super().__init__(pi_device=awg, channel_name=channel_name)
+        self._awg = awg
+
+    ################################################################################################
+    # Public Methods
+    ################################################################################################
+    def set_waveform_properties(
+        self,
+        output_signal_path: Optional[SignalGeneratorOutputPathsBase],
+        waveform_name: str,
+        amplitude: float,
+        offset: float,
+    ) -> None:
+        """Set the given parameters on the provided source channel.
+
+        Args:
+            output_signal_path: The output signal path of the specified channel.
+            waveform_name: The name of the waveform from the waveform list to generate.
+            amplitude: The amplitude of the signal to generate.
+            offset: The offset of the signal to generate.
+        """
+        self.set_amplitude(amplitude)
+        self.set_output_signal_path(output_signal_path)
+        self.set_offset(offset)
+        self.load_waveform(waveform_name)
+
+    def set_amplitude(self, value: float, absolute_tolerance: float = 0) -> None:
+        """Set the amplitude on the source channel.
+
+        Args:
+            value: The amplitude value to set.
+            absolute_tolerance: The acceptable difference between two floating point values.
+        """
+        self._awg.set_if_needed(
+            f"{self.name}:VOLTAGE:AMPLITUDE",
+            value,
+            tolerance=absolute_tolerance,
+        )
+
+    def set_offset(self, value: float, absolute_tolerance: float = 0) -> None:
+        """Set the offset on the source channel.
+
+        Args:
+            value: The offset value to set.
+            absolute_tolerance: The acceptable difference between two floating point values.
+        """
+        output_path = self._awg.query(f"OUTPUT{self.num}:PATH?")
+        if output_path == self._awg.OutputSignalPath.DCA.value:
+            self._awg.set_if_needed(
+                f"{self.name}:VOLTAGE:OFFSET",
+                value,
+                tolerance=absolute_tolerance,
+            )
+        elif value:
+            # No error is raised when 0 is the offset value and the output signal path is
+            # in a state where offset cannot be set.
+            offset_error = (
+                f"The offset can only be set with an output signal path of "
+                f"{self._awg.OutputSignalPath.DCA.value}."
+            )
+            raise ValueError(offset_error)
+
+    def set_state(self, value: int) -> None:
```

Comment:
Any value in making this just accept a bool to do away with the type checking?

Reply:
You can, but it ends up looking strange: set_state(True)

it could be changed to like, channel_on, but that doesn't really reflect the scpi command, considering it's a 1 to 1 method